numba_cuda/numba/cuda/debuginfo.py

```diff
@@ -59,6 +59,33 @@ def _var_type(self, lltype, size, datamodel=None):
         # For other cases, use upstream Numba implementation
         return super()._var_type(lltype, size, datamodel=datamodel)
 
+    def _di_subroutine_type(self, line, function, argmap):
+        # The function call conv needs encoding.
+        llfunc = function
+        md = []
+
+        # Create metadata type for return value
+        if len(llfunc.args) > 0:
+            lltype = llfunc.args[0].type
```

Comment:
Does omitting the replacement of `.` with `$` compared to the original implementation not cause issues for GDB? I note the comment in the original version:

```python
name = llarg.name.replace('.', '$')    # for gdb to work correctly
```

I thought that a dot in a name might be problematic for it? (c.f. the same syntax for traversing the members of a struct or union)

Reply:
That `.` versioned name used to be written to the metadata node as the value of 'name' of DILocalVariable. But currently the `.` versioned variable nodes are merged to the unversioned / user symbol named metadata node. Therefore, they are no longer to be present in DWARF entry.
And when represented as member of a struct or union, their names in the metadata are also unversioned names with corresponding types, so there won't be issues omitting this original replacement.
In short, neither `.` nor `$` will show up in debugger.